rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -190,6 +194,7 @@ void TimeSource::on_parameter_event(const rcl_interfaces::msg::ParameterEvent::S
     if (it.second->value.bool_value) {
       parameter_state_ = SET_TRUE;
       enable_ros_time();
+      create_clock_sub();
     } else {
       parameter_state_ = SET_FALSE;
       disable_ros_time();
```

Comment:
Should the topic be destroyed here explicitly? (opposite of `create_clock_sub()`)

Reply:
I had reasoned that if the sim param had been set at some point then it wouldn't be so bad to leave the subscription around, but being symmetric might be better hey (so it reflects the same state a node that never had sim time set would have).

Reply:
Yeah, I'm not sure it will ever come up, but conversely I don't think it would be wrong to remove the subscription if the parameter were allowed to be changed during runtime (not sure where we landed on that point).

Reply:
yeah, it can be changed at runtime (there are parameter callbacks that will get triggered): https://github.com/ros2/rclcpp/pull/559/commits/9ddb1ae7faf2dbf5405b470cec1f456186284a9d